socorro/external/postgresql/connection_context.py

```diff
@@ -68,21 +81,41 @@ def __init__(self, config, local_config=None):
             local_config - this is the namespace within the complete config
                            where the actual database parameters are found"""
         super(ConnectionContext, self).__init__()
+
         self.config = config
+
         if local_config is None:
             local_config = config
+
+        if local_config['database_url']:
+            self.database_url = local_config['database_url']
+            url = urlparse.urlparse(local_config['database_url'])
+            local_config['database_username'] = url.username
+            local_config['database_password'] = url.password
+            local_config['database_hostname'] = url.hostname
+            local_config['database_port'] = url.port
+            local_config['database_name'] = url.path[1:]
+
+            self.db_kwargs = {
+                'host': url.hostname or 'localhost',
+                'user': url.username or os.environ.get('USER', 'test'),
+                'password': url.password or 'aPassword',
+                'port': url.port or 5432,
+                'database': url.path[1:] or 'postgres'
+            }
+
```

Comment:
You might want to add an `else:` here because `self.dsn` does not need to be created if you have `self.db_kwargs`